### PR TITLE
[checkpoint] Restructure fragment making progress

### PR DESCRIPTION
**Pull request by lxfind**
This PR makes slight adjustments to the fragment and checkpoint construction process:
1. Makes various timing longer. This helps reduce checkpoint logs a bit and makes the progress more obvious
2. Added more logging
3. In `create_fragments_and_make_checkpoint`, we only ask for proposals from validators that we don't have a local fragment yet. This avoids repeat constructing fragments that we have constructed before. We also do not attempt to make fragments up-front if `no_more_fragments` is set. Also, to make sure we always have a chance to construct checkpoint, we always try to construct checkpoint first, before making new fragments.
4. Removed `checkpoint_num_fragments_sent` metric. It's not meaningful

### CODE REVIEW

File: crates/sui-core/src/authority_active/checkpoint_driver/mod.rs [hunk 1]
```diff
@@ -85,10 +85,10 @@ impl Default for CheckpointProcessControl {
         CheckpointProcessControl {
             delay_on_quorum_failure: Duration::from_secs(10),
             delay_on_local_failure: Duration::from_secs(3),
-            long_pause_between_checkpoints: Duration::from_secs(60),
+            long_pause_between_checkpoints: Duration::from_secs(120),
             timeout_until_quorum: Duration::from_secs(60),
             extra_time_after_quorum: Duration::from_millis(200),
-            consensus_delay_estimate: Duration::from_secs(1),
+            consensus_delay_estimate: Duration::from_secs(2),
```
Comment (gdanezis) [inline crates/sui-core/src/authority_active/checkpoint_driver/mod.rs:91]: I honestly think we should have a delay estimation as in PR https://github.com/MystenLabs/sui/pull/3548

**Review by gdanezis — APPROVED** [review]
Great stuff -- good to see the incremental fragment construction!

File: crates/sui-core/src/authority_active/checkpoint_driver/mod.rs [hunk 2]
```diff
@@ -682,120 +674,136 @@ pub async fn create_fragments_and_make_checkpoint<A>(
...
-    // Exit when we have a checkpoint proposal.
-    let mut available_authorities: BTreeSet<AuthorityName> = committee
-        .voting_rights
-        .iter()
-        .map(|(name, _)| *name)
-        .collect();
-    available_authorities.remove(&active_authority.state.name); // remove ourselves
+    let mut available_authorities = committee.shuffle_by_stake(None, None);
+    // Remove ourselves and all validators that we have already diffed with.
+    let already_fragmented = checkpoint_db.lock().validators_already_fragmented_with();
```
[the remark this answers is not in the document]
Comment (lxfind) [reply crates/sui-core/src/authority_active/checkpoint_driver/mod.rs:683]: `validators_already_fragmented_with` obtains it from the local fragments table

File: crates/sui-core/src/authority_active/checkpoint_driver/mod.rs [hunk 3]
```diff
@@ -682,120 +674,136 @@ pub async fn create_fragments_and_make_checkpoint<A>(
...
+                    // Check if there is a latest checkpoint
+                    if let Some(prev) = prev_cert {
+                        if prev.summary.sequence_number >= next_checkpoint_sequence_number {
+                            // We are now behind, stop the process
+                            debug!(
+                                latest_cp_cert_seq=?prev.summary.sequence_number,
+                                expected_cp_seq=?next_checkpoint_sequence_number,
+                                "We are behind, abort checkpoint construction process",
+                            );
+                            break;
```
[the remark this answers is not in the document]
Comment (lxfind) [reply crates/sui-core/src/authority_active/checkpoint_driver/mod.rs:752]: Not sure if I understand, once we stop here, we would go back to the beginning of the big loop and start downloading the latest cert

**Pull request merged**